Tests/SentryProfilerTests/SentryProfileTestFixture.swift

```diff
@@ -96,39 +100,75 @@ class SentryProfileTestFixture {
+            self.readingsPerBatch = readingsPerBatch
+        }
+    }
+    var mockMetrics: MockMetric
+    
+    func setMockMetrics(_ mockMetrics: MockMetric) {
+        self.mockMetrics = mockMetrics
+        systemWrapper.overrides.cpuUsage = mockMetrics.cpuUsage
+        systemWrapper.overrides.memoryFootprintBytes = mockMetrics.memoryFootprint
+        systemWrapper.overrides.cpuEnergyUsage = 0
+        systemWrapper.overrides.cpuUsageError = nil
+        systemWrapper.overrides.memoryFootprintError = nil
+        systemWrapper.overrides.cpuEnergyUsageError = nil
+    }
     
 #if !os(macOS)
     // SentryFramesTracker starts assuming a frame rate of 60 Hz and will only log an update if it changes, so the first value here needs to be different for it to register.
     let mockFrameRateChangesPerBatch: [FrameRate] = [.high, .low, .high, .low]
     
     // Absolute timestamps must be adjusted per span when asserting
-    var expectedSlowFrames = [[String: Any]]()
-    var expectedFrozenFrames = [[String: Any]]()
-    var expectedFrameRateChanges = [[String: Any]]()
-    
-    func resetGPUExpectations() {
-        expectedSlowFrames = [[String: Any]]()
-        expectedFrozenFrames = [[String: Any]]()
-        expectedFrameRateChanges = [[String: Any]]()
+    var expectedTraceProfileSlowFrames = [[String: Any]]()
+    var expectedTraceProfileFrozenFrames = [[String: Any]]()
+    var expectedTraceProfileFrameRateChanges = [[String: Any]]()
+    
+    var expectedContinuousProfileSlowFrames = [[String: NSNumber]]()
+    var expectedContinuousProfileFrozenFrames = [[String: NSNumber]]()
+    var expectedContinuousProfileFrameRateChanges = [[String: NSNumber]]()
+    
+    func resetProfileGPUExpectations() {
+        expectedTraceProfileSlowFrames = [[String: Any]]()
+        expectedTraceProfileFrozenFrames = [[String: Any]]()
+        expectedTraceProfileFrameRateChanges = [[String: Any]]()
+        
+        expectedContinuousProfileSlowFrames = [[String: NSNumber]]()
+        expectedContinuousProfileFrozenFrames = [[String: NSNumber]]()
+        expectedContinuousProfileFrameRateChanges = [[String: NSNumber]]()
     }
 #endif // !os(macOS)
     
-    func gatherMockedMetrics(span: Span) throws {
-        // clear out any errors that might've been set in previous calls
-        systemWrapper.overrides.cpuUsageError = nil
-        systemWrapper.overrides.memoryFootprintError = nil
-        systemWrapper.overrides.cpuEnergyUsageError = nil
-        
-        // gather mocked metrics readings
-        for _ in 0..<mockUsageReadingsPerBatch {
+    lazy var df: DateFormatter = {
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ss.SSSZ"
+        return dateFormatter
+    }()
+    
+    func log(_ line: Int, _ message: String) {
+        print("\(df.string(from: Date())) [Sentry] [TEST] [\((#file as NSString).lastPathComponent):\(line)]: \(message)")
```

Comment:
`m`: What made you use `print` instead of `SentryLog.log` here?

Reply:
Hmm, i didn't even consider to use the sentry logger from tests lol 😅

Reply:
I think I'd prefer to keep them separate, at the very least, be able to have a separate instance of SentryLogger that can be used from test code, that doesn't interfere with the one in the SDK.